src/ui/components/shared/LoadingScreen.tsx

```diff
@@ -17,60 +18,80 @@ const Hoverboard = dynamic(() => import("./Hoverboard"), {
-            <Hoverboard color={hoverboardColor} />
+    <div className={styles.loadingScreenWrapper}>
+      <BubbleViewportWrapper>
+        <div className={styles.loadingScreenWrapper}>
+          <div className="flex flex-col items-center space-y-2">
+            <div className={styles.hoverboardWrapper} onClick={changeHoverboardColor}>
+              <Hoverboard color={hoverboardColor} />
+            </div>
+            {children}
           </div>
-          {children}
         </div>
-      </div>
-    </BubbleViewportWrapper>
+      </BubbleViewportWrapper>
+    </div>
   );
 }
 
+const phrases = [
+  "Tuning the flux capacitor...",
+  "Prepping DeLorean for time travel...",
+  "Revving up to 88 miles per hour...",
+  "Gathering 1.21 gigawatts of power...",
+  "Ensuring temporal paradox safeguards...",
+  "Being careful not to mess up your timeline...",
+];
+
 function LoadingScreen({
   uploading,
   awaitingSourcemaps,
   fallbackMessage,
-  stalledTimeout = 9000,
+  stalledTimeout = 15000,
 }: PropsFromRedux & { fallbackMessage: string; stalledTimeout?: number }) {
-  // The backend send events in this order: uploading replay -> uploading sourcemaps.
-  let waitingForMessage = <span>{fallbackMessage}</span>;
-  if (awaitingSourcemaps) {
-    waitingForMessage = <span>Uploading sourcemaps...</span>;
-    stalledTimeout = Infinity;
-  } else if (uploading) {
-    waitingForMessage = <span>Uploading {Math.round(+uploading.amount)}Mb</span>;
-    stalledTimeout = Infinity;
-  }
-
-  let [stalled, setStalled] = useState<boolean>(false);
-  useEffect(() => {
-    if (stalledTimeout === Infinity) {
-      return;
-    }
+  const longWaitMessage =
+    "<div><p>This is taking longer than expected.</p><p><a href='http://replay.io/discord' target='discord'>Contact us on Discord</a></p>";
 
-    const timeout = setTimeout(() => setStalled(true), stalledTimeout);
+  const [message, setMessage] = useState(fallbackMessage);
 
-    return () => clearTimeout(timeout);
-  }, [fallbackMessage, stalledTimeout]);
+  useEffect(() => {
+    const changeMessage = () => {
+      const randomIndex = Math.floor(Math.random() * phrases.length);
+      setMessage(phrases[randomIndex]);
+    };
+    const phraseTimeout = setTimeout(changeMessage, 5000);
+    // swap to cutesy phrase after 5 seconds
+    // note: this should only be called a single time
+    return () => clearTimeout(phraseTimeout);
+  }, []);
+
+  useEffect(() => {
+    const stalledTimeoutId = setTimeout(() => {
+      // after 15 seconds, switch to stalled message
+      setMessage(longWaitMessage);
+    }, stalledTimeout);
+    return () => clearTimeout(stalledTimeoutId);
+  }, [stalledTimeout]);
 
+  const waitingForMessage =
+    awaitingSourcemaps || uploading ? (
+      <span>Uploading {Math.round(uploading?.amount ? Number(uploading.amount) : 0)}Mb</span>
+    ) : (
+      <span dangerouslySetInnerHTML={{ __html: message }}></span>
+    );
```

Comment:
async PR comment: This (the `waitingForMessage` message) is no longer being used (and should be deleted). That would also mean the `awaitingSourcemaps` and `uploading` props are not being used, so we could remove them (and the Redux `connect()` HOC) entirely.